src/buckets/mod.ts

```diff
@@ -0,0 +1,73 @@
+import { S3Config } from "../config/mod.ts";
+import { GlobalConfig, SwiftConfig } from "../config/types.ts";
+
+export interface BucketStore {
+  buckets: Bucket[];
+}
+
+export class Bucket {
+  constructor(
+    private _name: string,
```

Comment:
no need for `_` prefix in typescript usually (rather python), you have use instead `#` to mark the variable really private